vllm/attention/backends/flash_attn.py

```diff
@@ -252,10 +253,11 @@ def _add_seq_group(
             # only allowing multiple of block_size chunk size.
             # NOTE: This only works for oooooooxxx style attention.
             block_table = []
-            if inter_data.prefix_cache_hit:
+            if inter_data.prefix_cache_hit or self.has_prefix_cache_hit:
```

Comment:
Does this bug only apply to FlashAttn but not other backends?

Also nit:
```suggestion
            self.has_prefix_cache_hit |= inter_data.prefix_cache_hit
            if self.has_prefix_cache_hit:
```

Reply:
done

Reply:
I meant do other backends have this bug as well? I saw that your added a test with FlashInfer backend, so if that test is passed I guess it's all good?

Reply:
yes I meant XFORMERS and FlashInfer are not impacted by this prefix cache hit situation. I added both to the test.